development-styleguide.md

```diff
@@ -0,0 +1,107 @@
+## 👨‍💻 DEVELOPMENT STYLE GUIDE
+
+A practical reason to use a style guide is that they help you write code.
+Furthermore, they show commons mistakes or scenario and how to handle them
+
+### Typescript
+
+#### Object modification. Why you should avoid it.
+
+- Due to its dynamic nature, JavaScript makes it extremely easy to modify objects that you do not own
+- Such practice should generally be avoided as it can lead to maintenance problems and can produce hard to find bugs
+- If you don't own it, don't modify it
+
+side effects, example
+
+```typescript
+type Person = {
+  name: string;
+  surname: string;
+  age: number;
+};
+
+function printNameMutation(p: Person) {
+  console.log(`${p.name} ${p.surname}`);
+  // this object mutation could broken software logic
+  p.name = "Rob";
+}
+
+function printName(p: Person) {
+  console.log(`${p.name} ${p.surname}`);
+}
+
+const david: Person = { name: "David", surname: "Smith", age: 20 };
+printNameMutation(david); // David Smith
+// the object has been mutated
+printName(david); // Rob Smith
+```
+
+### [FP-TS](https://gcanti.github.io/fp-ts/modules/Option.ts.html)
+
+#### [Option](https://gcanti.github.io/fp-ts/modules/Option.ts.html)
+
+```typescript
+import { Option, some } from "fp-ts/lib/Option";
+
+// this is a common scenario where we want to model an user with email preferences
+
+// undefined -> user doesn't give any feedback
+// true -> users wants to receive email
+// false -> users doesn't want to receive email
+type EmailPreference = boolean | undefined;
+
+type User = {
+  id: number;
+  emailPreference: EmailPreference;
+};
+
+const user: User = { id: 1, emailPreference: true };
+
+// check if user wants to receive email
+if (user.emailPreference !== undefined && user.emailPreference === true) {
+  // send email
+} else {
+  console.log("can't send email due to the user's preference");
+}
+
+// use option to avoid if/else
+type UserO = {
+  id: number;
+  emailPreference: Option<boolean>;
```

Comment:
I'd rather avoid to introduce `UserO`  and just call

`fromNullable(user.emailPreference).map(...)` to replace the if/then part

moreover I'd add a link to this example https://github.com/gcanti/fp-ts/blob/1.12.0/src/Option.ts#L150
which shows how to extract deep nested nullable values